src/main/java/org/eclipse/pass/schema/SchemaServlet.java

```diff
@@ -47,22 +88,54 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     protected void doPost(HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException {
 
-        // setCommonHeaders()
+        response.setHeader("Accept-Post", "application/json, text/plain");
+        response.setHeader("Server", "PASS schema service");
 
         // Create SchemaService instance to handle business logic
+        BufferedReader br = new BufferedReader(new InputStreamReader(request.getInputStream()));
 
-        // Encode resulting schema(s) into a JSON response object
+        if (request.getContentType() == "text/plain") {
+            repository_list = readText(br);
+        } else {
+            try {
+                repository_list = readJson(br);
+            } catch (Exception e) {
```

Comment:
Generally should avoid catching all exceptions like this. Also must log in printStackTrace